main.py

```diff
@@ -47,35 +53,86 @@ def start_server(self, socket=os_socket):
                                 break
                             message = json.loads(data)
                             print()
-                            print(f"Received by {data}")
-                            logger.debug(f"Received by {message['sender']}: {message}")
-                            self.message_queue.put(data)
 
-                            if not self.message_queue.empty():
-                                msg = self.message_queue.get()
+                            if message.get("type") == "GET_NODES":
+                                conn.sendall(json.dumps({"nodes": self.peer_hosts}).encode())
+                                new_peer = message.get("node_ip")
+                                self.peer_hosts.append(new_peer)
+                                print(f"server connected to {self.peer_hosts}")
+                                self.send_peers(peer_port)
+                            elif message.get("type") == "NEW_NODES":
+                                all_peers = set(self.peer_hosts)
+                                new_peers = message.get("nodes", [])
+                                all_peers.update(new_peers)
+                                all_peers.remove(self.ip)
+                                self.peer_hosts = list(all_peers)
+                            else:
+                                print(f"Received by {data}")
+                                logger.debug(f"Received by {message['sender']}: {message}")
+                                self.message_queue.put(data)
+
+                                if not self.message_queue.empty():
+                                    msg = self.message_queue.get()
 
-                                ack = json.dumps({"type": "ack", "message": f"Received {message['message']} from {message['sender']}", "sender": self.nickname}).encode()
-                                conn.sendall(ack)
-                                logger.debug(f"{self.nickname} sent ack {ack}")
+                                    ack = json.dumps({"type": "ack", "message": f"Received {message['message']} from {message['sender']}", "sender": self.nickname}).encode()
+                                    conn.sendall(ack)
+                                    logger.debug(f"{self.nickname} sent ack {ack}")
         except Exception as exc:
             logger.exception(exc)
             raise exc
 
+    def request_peers(self, peer_port, socket=os_socket):
+
+        all_peers = set(self.peer_hosts)
+        try:
+            with socket(AF_INET, SOCK_STREAM) as s:
+                s.connect(("startup_server", peer_port))
```

Comment:
This won't be very scalable. Can we just do the request all to peers that we received from commandline arguments